### PR TITLE
update dependencies and cleanup gradle files

### PR DESCRIPTION
**Pull request by FabianTerhorst**
update okhttp to a stable version
update lombok

### CODE REVIEW

**Comment by FabianTerhorst** [comment]
The protobuf update would require differently generated protofiles.

**Comment by FabianTerhorst** [comment]
travis is happy

**Comment by FabianTerhorst** [comment]
@Grover-c13 @jabbink ready to merge. Its also including the change we discussed on slack.

File: build.gradle [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
-    }
-}
-
 allprojects {
+    apply plugin: 'java'
+    group = 'com.pokegoapi'
+    version = '0.3-1'
+    sourceCompatibility = 1.7
+    targetCompatibility = 1.7
+    archivesBaseName = 'PokeGOAPI-sample'
```
Comment (AbandonedCart) [inline build.gradle]: You are naming every build `sample` including the library.

File: build.gradle [hunk 2]
```diff
@@ -1,14 +1,11 @@
...
-    repositories {
-    }
-    dependencies {
-    }
-}
-
 allprojects {
+    apply plugin: 'java'
+    group = 'com.pokegoapi'
+    version = '0.3-1'
```
Comment (AbandonedCart) [inline build.gradle]: This does not match the version number you provided in https://github.com/Grover-c13/PokeGOAPI-Java/pull/453. There needs to be consistency in versioning.

Comment (FabianTerhorst) [inline build.gradle]: why you comment on outdated divs?

**Comment by AbandonedCart** [comment]
It isn't ready to merge. See the notes attached to https://github.com/Grover-c13/PokeGOAPI-Java/pull/448/commits/4bd9f2aeffe69d78e1e063fa698dcec4101b2f67

**Comment by FabianTerhorst** [comment]
@LoungeKatt archivesBaseName isn´t required in sample.

**Comment by FabianTerhorst** [comment]
Don´t comment on outdated divs. Always refresh the site before.

**Comment by AbandonedCart** [comment]
@FabianTerhorst That is not what I said. You had renamed the entire project `sample` meaning you would have sample-sources, sample-javadoc, sample-all, sample (the library without dependencies), sample. The next commit fixed it, but now sample will only be named `sample` and not include the project name.

I can understand your frustrated that I keep correcting you, but then stop being hasty.

**Comment by FabianTerhorst** [comment]
@LoungeKatt sample isn´t published to jcenter. You know any case where archivesBaseName is needed for an sample?

**Comment by AbandonedCart** [comment]
@FabianTerhorst for compiling it locally. Is there any reason not to append the name to avoid a folder full of files named `library` and `sample` that do not identify the project once they are moved out of the build folder?

**Comment by FabianTerhorst** [comment]
maybe start using http://semver.org/ for version numbers?

File: library/build.gradle [hunk 3]
```diff
@@ -33,7 +33,7 @@ processResources {
 
 // Run this task to bundle all needed dependency
 task bundle(type: Jar) {
-    baseName = archivesBaseName + '-all'
+    baseName = archivesBaseName + '-' + version + '-all'
```
Comment (AbandonedCart) [inline library/build.gradle]: It already does this by default when building the jar. This will result in NAME-VERSION-VERSION-all

**Comment by AbandonedCart** [comment]
@FabianTerhorst If you prefer, I can add the necessary changes for what you mean to accomplish to the README PR. It will save about 10 commits of trial and error.

You may want to try testing things before you commit them to a live PR. Just a thought.

File: build.gradle (hunk 2 above)
Comment (AbandonedCart) [inline build.gradle]: You needed the version number updated just to turn around and change it later. This changed again later in this very pull. They are version numbers, not tags. They change with a new release, not current mood.

**Comment by FabianTerhorst** [comment]
i squash the commits and reopen the pr

**Comment by FabianTerhorst** [comment]
#458

**Comment by AbandonedCart** [comment]
@FabianTerhorst Thank you. That is much less tedious to read.

**Pull request closed without merging**